src/components/calcite-rating/calcite-rating.tsx

```diff
@@ -0,0 +1,366 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Listen,
+  Method,
+  Prop,
+  State
+} from "@stencil/core";
+import { getKey } from "../../utils/key";
+import { getElementDir, hasLabel } from "../../utils/dom";
+
+@Component({
+  tag: "calcite-rating",
+  styleUrl: "calcite-rating.scss",
+  shadow: true
+})
+export class CalciteRating {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteRatingElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /** specify the theme of scrim, defaults to light */
+  @Prop({ reflect: true }) theme: "light" | "dark";
+
+  /** specify the scale of the component, defaults to m */
+  @Prop({ reflect: true }) scale: "s" | "m" | "l" = "m";
+
+  /** the value of the rating component */
+  @Prop({ reflect: true }) value = 0;
+
+  /** is the rating component in a selectable mode */
+  @Prop({ reflect: true }) readOnly = false;
```

Comment:
Are we keeping both readOnly and disabled?

Reply:
I feel like they are separate use cases?

Read-only: Useful if you want to show that a rating has been entered and perhaps not able to be changed by a user (submitting a review on ecommerce)

Disabled: Useful if the rating isn't available at this time to a user. Perhaps if the rating module needs to be disabled while a user is editing another portion of an item or isn't available yet.

https://bootstrap-vue.org/docs/components/form-rating#disabled
https://bootstrap-vue.org/docs/components/form-rating#readonly

The display is also a little different - readonly doesn't show the "could be filled affordance".